vaadin-text-field-flow-parent/vaadin-text-field-flow-integration-tests/src/test/java/com/vaadin/flow/component/textfield/tests/TextAreaPatternIT.java

```diff
@@ -0,0 +1,132 @@
+ */
+@TestPath("vaadin-text-field/text-area-pattern")
+public class TextAreaPatternIT extends AbstractComponentIT {
+
+    private TextAreaElement textArea;
+    private TestBenchElement currentValue;
+    private TestBenchElement setNumberPattern;
+    private TestBenchElement setInvalidPattern;
+    private TestBenchElement clearPattern;
+    private TestBenchElement checkValidity;
+    private TestBenchElement validityOutput;
+    private TestBenchElement enablePreventInvalidInput;
+
+    @Before
+    public void init() {
+        open();
+        textArea = $(TextAreaElement.class).first();
+        currentValue = $("div").id("current-value");
+        setNumberPattern = $("button").id("set-number-pattern");
+        setInvalidPattern = $("button").id("set-invalid-pattern");
+        clearPattern = $("button").id("clear-pattern");
+        enablePreventInvalidInput = $("button")
+                .id("enable-prevent-invalid-input");
+        // These two are defined by TextFieldTestPageUtil.addInvalidCheck
+        checkValidity = $("button").id("check-is-invalid");
+        validityOutput = $("div").id("is-invalid");
+    }
+
+    private void assertClientSideValid() {
+        Assert.assertFalse(textArea.hasAttribute("invalid"));
+    }
+
+    private void assertClientSideInvalid() {
+        Assert.assertTrue(textArea.hasAttribute("invalid"));
+    }
+
+    private void assertServerSideValid() {
+        checkValidity.click();
+        Assert.assertEquals("valid", validityOutput.getText());
+    }
+
+    private void assertServerSideInvalid() {
+        checkValidity.click();
+        Assert.assertEquals("invalid", validityOutput.getText());
+    }
+
+    @Test
+    public void validPattern_validInput_isValid() {
+        setNumberPattern.click();
+        textArea.sendKeys("1234");
+        blur();
+
+        assertClientSideValid();
+        assertServerSideValid();
+    }
+
+    @Test
+    public void validPattern_invalidInput_isInvalid() {
+        setNumberPattern.click();
+        textArea.sendKeys("abcd");
+        blur();
+
+        assertClientSideInvalid();
+        assertServerSideInvalid();
+    }
+
+    @Test
+    public void validPattern_preventInvalidInput_invalidInput_inputShouldBePrevented() {
+        setNumberPattern.click();
+        enablePreventInvalidInput.click();
+        textArea.sendKeys("abcd1234");
+        blur();
+
+        assertClientSideValid();
+        assertServerSideValid();
+        Assert.assertEquals("1234", currentValue.getText());
+    }
+
+    @Test
+    public void invalidPattern_anyInput_isValid() {
+        setInvalidPattern.click();
+        textArea.sendKeys("abcd");
+        blur();
+
+        assertClientSideValid();
+        assertServerSideValid();
+    }
+
+    // Ignored: Currently changing a constraint on a Flow component does not
+    // trigger validation, so the component stays invalid even after clearing a
+    // constraint.
+    @Test
+    @Ignore
```

Comment:
A link to the issue would be good to have. A few other ignored tests use the following approach:

```suggestion
    @Ignore("Missing logic, see https://github.com/vaadin/flow-components/issues/2370")
```

Reply:
Nice, didn't know about that. Done.